lib/rex/proto/kerberos/crypto/gss_new_encryption_type.rb

```diff
@@ -0,0 +1,130 @@
+module Rex
+  module Proto
+    module Kerberos
+      module Crypto
+        # The RFC4121 implementation of GSS wrapping (Section 4.2.6.2)
+        # which applies to "newer encryption types" (defined as those not
+        # in section 1 of RFC4121)
+        # This mixin may be included by Encryption providers in Rex::Proto::Kerberos::Crypto
+        module GssNewEncryptionType
+          GSS_SENT_BY_ACCEPTOR = 1
+          GSS_SEALED = 2
+          GSS_ACCEPTOR_SUBKEY = 4
+
+          # The length of the GSS header
+          GSS_HEADER_LEN = 16
+
+          #
+          # Encrypt the message, wrapping it in GSS structures
+          # @return [String, Integer, Integer] The encrypted data, the length of its header, and the length of padding added to it prior to encryption
+          #
+          def gss_wrap(plaintext, key, sequence_number, is_initiator, use_acceptor_subkey: true)
+            flags = GSS_SEALED
+            flags |= GSS_ACCEPTOR_SUBKEY if use_acceptor_subkey
+
+            if is_initiator
+              key_usage = Rex::Proto::Kerberos::Crypto::KeyUsage::GSS_INITIATOR_SEAL
+            else
+              key_usage = Rex::Proto::Kerberos::Crypto::KeyUsage::GSS_ACCEPTOR_SEAL
+              flags |= GSS_SENT_BY_ACCEPTOR
+            end
+
+            tok_id = 0x0504
+            filler = 0xFF
+            ec = calculate_ec(plaintext)
+            rrc = calculate_rrc
+
+            # RFC4121, Section 4.2.4
+            plaintext_header = [tok_id, flags, filler, 0, 0, sequence_number].pack('nCCnnQ>')
+
+            header = [tok_id, flags, filler, ec, rrc, sequence_number].pack('nCCnnQ>')
+            ec_filler = "x" * ec
```

Comment:
@smashery Is there any context on this, is `x` the normal value here? 👀

Reply:
I couldn't find an official definition of what this value should be. RFC4121 says "The
 values and size of the filler octets are chosen by implementations, such that there SHALL be no crypto-system residue present after the
 decryption." (https://datatracker.ietf.org/doc/html/rfc4121#section-4.2.4)

I used `'x'` based on the [(Linux implementation of the Kerberos client)](https://salsa.debian.org/debian/krb5/-/blob/0269810b1aec6c554fb746433f045d59fd34ab3a/src/lib/gssapi/krb5/k5sealv3.c#L160)

I'll make that clearer in a comment.